esphome/components/opentherm/switch/__init__.py

```diff
@@ -0,0 +1,72 @@
+import esphome.codegen as cg
+import esphome.config_validation as cv
+from esphome.components import switch
+from esphome.const import (
+    ICON_RADIATOR,
+)
+from ...opentherm import (
+    OpenThermComponent,
+    CONF_OPENTHERM_ID,
+)
+from .. import opentherm
+
+CustomSwitch = opentherm.class_("CustomSwitch", switch.Switch, cg.Component)
```

Comment:
These classes really should be named `OpenThermXXX`

```suggestion
OpenThermSwitch = opentherm.class_("OpenThermSwitch", switch.Switch, cg.Component)
```